src/state/trajectory/logics.ts

```diff
@@ -345,6 +353,102 @@ const loadFileViaUrl = createLogic({
+            controller = new SimulariumController({
+                netConnectionSettings: netConnectionConfig,
+            });
+            dispatch(setSimulariumController(controller));
+        }
+        // check the server health
+        // Originally thought to send checks every 15 seconds,
+        // currently sending a flurry of checks, 3 seconds apart, 15 seconds total.
+        // If any come back true we assume we're good for now... this timing is arbitrary
+        // I reasoned a quick set of requests that can be started on page load,
+        // that cumulatively have enough delay to let a connection be configured
+        // is a better idea than a minute long period of sending checks,
+        // during which things could change.
+        // We can retrigger this as often as necessary.
+        let healthCheckSuccessful = false;
+        const healthCheckTimeouts: HealthCheckTimeout = {};
+        const attempts = 0;
+
+        // recursive function that sends response handlers to viewer with request and timeout ids
+        const performHealthCheck = (attempts: number) => {
+            if (healthCheckSuccessful) {
+                return; // Stop if a successful response was already received
+            }
+            const MAX_ATTEMPTS = 5;
+            const requestId: string = uuidv4();
+
+            controller.checkServerHealth(() => {
+                // callback/handler for viewer function
+                healthCheckSuccessful = true;
+                clearTimeout(healthCheckTimeouts[requestId]);
+                dispatch(
+                    setConversionStatus({
+                        status: CONVERSION_SERVER_LIVE,
+                    })
+                );
+                done();
+            }, netConnectionConfig);
+
+            // timeouts that, if they resolve, send new checks until the max # of attempts is reached
+            const timeoutId = setTimeout(() => {
+                if (!healthCheckSuccessful) {
+                    // in case another check just resolved
+                    clearTimeout(healthCheckTimeouts[requestId]);
+                    if (attempts < MAX_ATTEMPTS) {
+                        // retry the health check with incremented count
+                        attempts++;
+                        performHealthCheck(attempts);
+                    } else {
+                        // if we've done the max # of attempts, set conversionStatus
+                        dispatch(
+                            setConversionStatus({
+                                status: CONVERSION_NO_SERVER,
+                            })
+                        );
+                        done();
+                    }
+                }
+            }, 3000);
+
+            // store the time out id
+            healthCheckTimeouts[requestId] = timeoutId;
+        };
+
+        // Start the first health check
+        performHealthCheck(attempts);
+    },
+    type: CONFIGURE_FILE_CONVERSION,
```

Comment:
or do we want to just run `CONVERT_FILE` with no payload whenever we health check? that's less actions/logics but also less semantic to me

Reply:
when we go to the `/import` route to show the form we just trigger these:
```
                dispatch(setIsPlaying(false));
                dispatch(setStatus({ status: VIEWER_IMPORTING }));
 ```
 
 the `convertFile()` action isn't called anywhere yet, but I assumed that's what we would use to send the uploaded file to octopus

Reply:
Ok yes, I used `RECEIVE_FILE_TO_CONVERT` for readability since it doesn't have it's own redux logic.

Added some safeguards in reducer so that we can call with `null` payload on page load, but if I'm just being stubborn and you really only want this logic done as a side effect of other actions I'll remove it.